speeduino/sensors.ino

```diff
@@ -113,22 +112,143 @@ void initialiseADC()
-  //Each sensor has it's own default value
-  if(configPage4.ADCFILTER_TPS  > 240) { configPage4.ADCFILTER_TPS   = ADCFILTER_TPS_DEFAULT;   writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_CLT  > 240) { configPage4.ADCFILTER_CLT   = ADCFILTER_CLT_DEFAULT;   writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_IAT  > 240) { configPage4.ADCFILTER_IAT   = ADCFILTER_IAT_DEFAULT;   writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_O2   > 240) { configPage4.ADCFILTER_O2    = ADCFILTER_O2_DEFAULT;    writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_BAT  > 240) { configPage4.ADCFILTER_BAT   = ADCFILTER_BAT_DEFAULT;   writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_MAP  > 240) { configPage4.ADCFILTER_MAP   = ADCFILTER_MAP_DEFAULT;   writeConfig(ignSetPage); }
-  if(configPage4.ADCFILTER_BARO > 240) { configPage4.ADCFILTER_BARO  = ADCFILTER_BARO_DEFAULT;  writeConfig(ignSetPage); }
-  if(configPage4.FILTER_FLEX    > 240) { configPage4.FILTER_FLEX     = FILTER_FLEX_DEFAULT;     writeConfig(ignSetPage); }
+/** ADC sequencer.
+ *  Should be called repeatedly.
+ *  Starts ADC coversions in sequence, 
+ *  deals with checking completed conversions and reads the results.
+ */
+void ADC_sequencer(){
 
-  flexStartTime = micros();
+static enum ADCoperations:uint8_t {MAPadc,EMAPadc,TPSadc,IATadc,O2adc,O2_2adc,BATadc,CLTadc,FuelpressureAdc,OilpressureAdc,BaroAdc} adcOperation =MAPadc; //current state of ADCsequence,initial state is MAPadc
+static ADCstates adcState = ADCidle;  //current state of ADC,initial state is idle
+static byte adcLoopTimer=0;           //holds timer flags
 
-  vssIndex = 0;
+adcLoopTimer |= LOOP_TIMER; //pick up timer bits as they are set, this can only set bits, not clear
+
+if (ADC_CheckForConversionComplete() == true)
+{
+  adcState=ADCcomplete;
+}
+while(adcState !=ADCrunning) //do not leave the scene until we have gotten the ADC to run, also do nothing when ADC is still in progress
+  {
+    switch (adcOperation)
+    {
+    case MAPadc:
+      adcState = readMAP(adcState);
+      if (adcState == ADCidle)
+      {                          //when this channel done
+        adcOperation = EMAPadc; //specify next operation
+      }
+      break;
+    case EMAPadc:
+      adcState = readEMAP(adcState);
+      if (adcState == ADCidle)
+      {                         //when this channel done
+        adcOperation = TPSadc; //specify next operation
+      }
+      break;
+    case TPSadc:                                    //40Hz
+      if(BIT_CHECK(adcLoopTimer, BIT_TIMER_40HZ)) //(any faster and it can upset the TPSdot sampling time)
+      {
+        adcState = readTPS(true, adcState); //read TPS        
+        if (adcState == ADCidle)
+        {                         //when this channel done
+          readTPSdot();
+          adcOperation = IATadc;  //specify next operation
+          BIT_CLEAR(adcLoopTimer, BIT_TIMER_40HZ);
```

Comment:
You should clear the `TIMER_mask` here, as `adcLoopTimer` will make a copy of it at next loop. Look at main loop for reference.